ui/src/pageLayout/components/UserWidget.tsx

```diff
@@ -0,0 +1,119 @@
+// Libraries
+import React, {FC} from 'react'
+import {Link} from 'react-router'
+import {connect} from 'react-redux'
+
+// Components
+import {TreeNav} from '@influxdata/clockface'
+import CloudExclude from 'src/shared/components/cloud/CloudExclude'
+import CloudOnly from 'src/shared/components/cloud/CloudOnly'
+
+// Actions
+import {showOverlay, dismissOverlay} from 'src/overlays/actions/overlays'
+
+// Constants
+import {
+  CLOUD_URL,
+  CLOUD_LOGOUT_PATH,
+  CLOUD_USAGE_PATH,
+  CLOUD_BILLING_PATH,
+} from 'src/shared/constants'
+
+// Types
+import {AppState, Organization} from 'src/types'
+import {MeState} from 'src/shared/reducers/me'
+
+// Selectors
+import {getOrg} from 'src/organizations/selectors'
+
+interface StateProps {
+  org: Organization
+  me: MeState
+}
+
+interface DispatchProps {
+  handleShowOverlay: typeof showOverlay
+  handleDismissOverlay: typeof dismissOverlay
+}
+
+type Props = StateProps & DispatchProps
+
+const UserWidget: FC<Props> = ({
+  org,
+  me,
+  handleShowOverlay,
+  handleDismissOverlay,
+}) => {
+  const logoutURL = `${CLOUD_URL}${CLOUD_LOGOUT_PATH}`
+
+  if (!org) {
+    return null
+  }
+
+  const handleSwitchOrganizations = (): void => {
+    handleShowOverlay('switch-organizations', {}, handleDismissOverlay)
+  }
+
+  return (
+    <TreeNav.User username={me.name} team={org.name}>
+      <CloudOnly>
+        <TreeNav.UserItem
+          id="usage"
+          label="Usage"
+          linkElement={className => (
+            <a className={className} href={CLOUD_USAGE_PATH} />
+          )}
+        />
+        <TreeNav.UserItem
+          id="billing"
+          label="Billing"
+          linkElement={className => (
+            <a className={className} href={CLOUD_BILLING_PATH} />
+          )}
+        />
+        <TreeNav.UserItem
+          id="logout"
+          label="Logout"
+          linkElement={className => (
+            <a className={className} href={logoutURL} />
+          )}
+        />
+      </CloudOnly>
+      <CloudExclude>
```

Comment:
I'd be interested in seeing what would happen if we used the same `Logout` component for CLOUD and OSS since the `Logout` component calls the same `logoutUrl` if it's on "Cloud"

Reply:
I'm gonna wrap this new navbar in a feature flag so we can test it out in acceptance, maybe that is something we can test. Would be nice to reduce separation between cloud/oss as much as possible